### PR TITLE
Design Document, add ID_STAGE description

### PR DESCRIPTION
**Pull request by JeanRochCoulon**
Add chapters corresponding to the N-1 CVA6 hierarchy.
Add block diagrams.
Add ID_STAGE first description, to be fine-tuned.

### CODE REVIEW

File: docs/04_cv32a65x_design/source/cva6_id_stage.rst [hunk 1]
```diff
@@ -15,15 +15,68 @@ ID_STAGE Module
...
+   ID_STAGE submodules
+
+
+Compressed_decoder
+~~~~~~~~~~~~~~~~~~
+
+The compressed_decoder module decompresses all the compressed
+instructions taking a 16-bit compressed instruction and expanding it
+to its 32-bit equivalent.
+All compressed instructions have a 32-bit equivalent.
```
Comment (jquevremont) [inline docs/04_cv32a65x_design/source/cva6_id_stage.rst:62]: This is true with CVA6 (so no change is required).
FYI: This is no more true globally for the RISC-V ISA as the Zcmp extension includes instructions like push and pop that are expanded as series of 32-bit instructions. And Zcmt instructions have no equivalent 32-bits instructions.

File: docs/04_cv32a65x_design/source/mmu.rst [hunk 2]
```diff
@@ -0,0 +1,1580 @@
+.. _CVA6_MMU:
+
+
+----------------------
+Memory Management Unit
```
Comment (jquevremont) [inline docs/04_cv32a65x_design/source/mmu.rst:5]: MMU in the embedded CV32A65X?
FYI, PlanV is upgrading the MMU documentation to expand with Sv39 and and Sv39x4, as part of the MMU unification.

File: docs/04_cv32a65x_design/source/subsystem.rst [hunk 3]
```diff
@@ -26,7 +26,12 @@ At least 6 cycles are needed to execute one instruction.
 Connection with other sub-systems
```
Comment (jquevremont) [inline docs/04_cv32a65x_design/source/subsystem.rst:26]: At line 18, missing period at the end of the sentence.
At line 21 "The processor is single issue", no more true for CV32A65X.

**Review by jquevremont — APPROVED** [review]
Thanks for the huge effort. Very minor comments (no need for me to re-review).

File: docs/04_cv32a65x_design/source/port_ras.rst [hunk 4]
```diff
@@ -15,7 +15,7 @@
    * - Signal
      - IO
      - Description
-     - Connection
+     - connexion
```
Comment (jquevremont) [inline docs/04_cv32a65x_design/source/port_ras.rst:18]: "Connexion" in French, "connection" in English. This is counter-intuitive.

**Review by jquevremont — APPROVED** [review]

File: docs/04_cv32a65x_design/source/port_ras.rst (hunk 4 above)
Comment (JeanRochCoulon) [reply docs/04_cv32a65x_design/source/port_ras.rst:18]: Oeufs corses !   Fixed

File: docs/04_cv32a65x_design/source/mmu.rst (hunk 2 above)
Comment (JeanRochCoulon) [reply docs/04_cv32a65x_design/source/mmu.rst:5]: MMU removed from 65X DD

File: docs/04_cv32a65x_design/source/subsystem.rst (hunk 3 above)
Comment (JeanRochCoulon) [reply docs/04_cv32a65x_design/source/subsystem.rst:26]: Thanks for the feedback, I will fix in a next PR.

**Pull request merged**